csharp/src/Apache.Arrow/Arrays/BinaryArray.cs

```diff
@@ -48,21 +48,28 @@ public BinaryArray(ArrowTypeId typeId, ArrayData data)
             data.EnsureBufferCount(3);
         }
 
-        public abstract class BuilderBase<TArray, TBuilder> : IArrowArrayBuilder<byte, TArray, TBuilder>
+        public abstract class BuilderBase<T, TArray, TBuilder> : IArrowArrayBuilder<T, TArray, TBuilder>
```

Comment:
Why did we need to add the `T` here? This is the `BinaryArray` class, which implies the `T` is `byte`. And looking at all the usages (above and in `StringArray`) we are only filling the `T` with `byte`.

Reply:
Early on in the effort I was trying to get a more generalized BuilderBase, and one that could be more easily re-used by end users (such as myself) who might be interested in implementing a custom type. It didn't really end working out that way, however, so I have reverted it back.